cincinnati/src/plugins/internal/edge_add_remove.rs

```diff
@@ -205,6 +221,29 @@ impl EdgeAddRemovePlugin {
     }
 }
 
+/// Try to find the architecture metadata and add it to the version String assuming SemVer.
+///
+/// If the referenced ReleaseId doesn't have the arch metadata, the version
+/// string will be passed through unchanged.
+fn try_annotate_semver_build(
```

Comment:
As in [this comment][1], I'd rather draw edges to all matching versions regardless of arch, and let the arch-plugin go through and cull any cross-arch stuff at the end.  Easiest way to do that is probably to use a new `graph.iter_by_version(version: &str)` that iterates over all nodes and returns any where the *original* version (before we inject the uniquifying arch into `build`) matches.  Then you can iterate over the result and add/remove edges to/from all matching releases. 

[1]: https://github.com/openshift/cincinnati/pull/176#discussion_r347744320

Reply:
For reasons explained in https://github.com/openshift/cincinnati/pull/176#discussion_r348570055 I'd like to keep the Cincinnati core library correct and do the fiddling in the plugins.